### PR TITLE
GitHub Action to automate the identification of common misspellings

### PR DESCRIPTION
**Pull request by cclauss**
GitHub Action to automate the identification of common misspellings in text files.
https://github.com/codespell-project/codespell

### CODE REVIEW

**Comment by cclauss** [comment]
This can be used by other projects to rapidly add __codespell__ to their repos.

**Comment by larsoner** [comment]
Rather than actually add this as an action for this repo, we could add this to the `README`. What you have here will fail because we have:
```
larsoner@bunk:~/python/codespell$ codespell | wc -l
...
186
```
So if I merge this PR it will fail the check. If you want to make it pass, then you'll need to tweak the call. For us maybe `codespell codespell_lib/_codespell.py` is good enough. That passes for me locally.

File: .github/workflows/codespell.yml [hunk 1]
```diff
@@ -9,4 +9,5 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v1
       - run: pip install codespell
-      - run: codespell
+      # When using this Action in other repos, the --skip option below can be removed
+      - run: codespell --skip=./.*,./codespell_lib,./example/code.c
```
Comment (larsoner) [inline .github/workflows/codespell.yml]: This now skis `codespell_lib` which is where the library code actually lives, which is not so helpful. Maybe:
```
codespell --skip=./.*,./example/code.c,*.pyc,./codespell_lib/data,test_basic.py
```

**Comment by larsoner** [comment]
Thanks @cclauss

**Pull request merged**